### ISSUE TITLE
Getting 404 for my username

### ISSUE THREAD

**Issue by bsethi** [blue-yonder/bonfire#30]
Getting this error:
```
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: <hostname>:<port>/users/<username>
```

This is after the execution of this line:

```
        r = requests.get(self.base_url + url, params=params, headers=self.get_header,
                         auth=(self.username, self.password), proxies=self.proxies)
```
in `graylog_api.py`

**Comment by valentin-krasontovitsch** [comment]
Hmm, a bit difficult to figure out just based on that - could you disclose your config file (if you have one), and the command you were running? I assume you've double checkd the url?

**Comment by bsethi** [comment]
Command that I'm running:
`bonfire -h x1.devfactory.com --port 13912 --stream 58638679143f0d0001658765 -u <username> -p <password> -@ "10 minutes ago" "*"`

Yes, I've checked the url, I get the logs on the WebUI at `http://x1.devfactory.com:13912/streams/58638679143f0d0001658765/search`

**Comment by valentin-krasontovitsch** [comment]
So printing out some variables from the line in `graylog_api.py` that you referred to, it seems like the request in question is for the url `http://x1.devfactory.com:13912/users/test-test` - if you could just curl that address from the same command line -

```
curl -Lv http://x1.devfactory.com:13912/users/<USERNAME>
```

that should give us a non 404 result, assuming that the user exists and that you are able to reach / communicate with the server... 

if that one doesn't work out, then I would say that the problem lies elsewhere, not with graylog. if you get a 200 result though... then we have to look more : )

**Comment by bsethi** [comment]
This is the response I get

```
*   Trying 10.69.11.89...
* TCP_NODELAY set
* Connected to x1.devfactory.com (10.69.11.89) port 13912 (#0)
> GET /users/<username> HTTP/1.1
> Host: x1.devfactory.com:13912
> User-Agent: curl/7.60.0
> Accept: */*
>
< HTTP/1.1 200 OK
< X-Graylog-Node-ID: 421b9e37-c2d9-4a15-83f9-468dc298d5d9
< X-UA-Compatible: IE=edge
< Content-Type: text/html
< Date: Sat, 09 Mar 2019 12:07:36 GMT
< Content-Length: 1290
<
<!DOCTYPE html>
<html>
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="robots" content="noindex, nofollow">
    <meta charset="UTF-8">
    <title>Graylog Web Interface</title>
    <link rel="shortcut icon" href="/assets/favicon.png">

  </head>
  <body>
    <script src="/config.js"></script>
    <script src="/assets/vendor.js"></script>

    <script src="/assets/polyfill.7c2b0dd11d39e6e4468c.js"></script>

    <script src="/assets/plugin/org.graylog.plugins.pipelineprocessor.ProcessorPlugin/plugin.org.graylog.plugins.pipelineprocessor.PipelineProcessorPlugin.8cb610e9be38ce39b2d9.js"></script>

    <script src="/assets/plugin/org.graylog.plugins.map.MapWidgetPlugin/plugin.org.graylog.plugins.map.MapWidgetPlugin.3889c9d8f130e7e01bc7.js"></script>

    <script src="/assets/plugin/org.graylog.plugins.enterprise_integration.EnterpriseIntegrationPlugin/plugin.org.graylog.plugins.enterprise_integration.EnterpriseIntegrationPlugin.2a15f3e3b09335a4acc6.js"></script>

    <script src="/assets/plugin/org.graylog.plugins.collector.CollectorPlugin/plugin.org.graylog.plugins.collector.CollectorPlugin.9bcb9081b7c0c3c86c7d.js"></script>

    <script src="/assets/app.7c2b0dd11d39e6e4468c.js"></script>

  </body>
</html>
* Connection #0 to host x1.devfactory.com left intact
```

**Comment by valentin-krasontovitsch** [comment]
oh ok that's hmm... I assume you're running it on the same server, and in the same console, with the same environment variables and whatnot, the curl and the bonfire command? 

gonna have to dig a bit deeper, and get back to you a bit later!

**Comment by bsethi** [comment]
Yes, both the commands are run in the same environment.

**Comment by bsethi** [comment]
Any updates @valentin-krasontovitsch ?

**Comment by valentin-krasontovitsch** [comment]
Very sorry about the late reply - so I'm still not super sure, and not certain, but I think I had you try a silly curl, because it was missing a header, namely `Accept: application/json`. Comparing with our graylog instance, I'm seeing the same result as you for the curl I gave you, but the following curl (which more accurately mirrors what bonfire does)

```
curl -Lv -H 'Accept: application/json' http://x1.devfactory.com:13912/users/<USERNAME>
```

returns a 404, just like bonfire claims.

Looking more precisely at what happens in my own bonfire setup with config file, I noticed that the url includes an `/api` up front - and that I supply that in my config file.

So the solution to your problem might be supplying an api endpoint as well - like such:

```
bonfire -h x1.devfactory.com --port 13912 --stream 58638679143f0d0001658765 \
  -u <username> -p <password> -@ "10 minutes ago" "*" --endpoint /api
```

Sadly, the [commit that introduced the endpoint option](#18) was not part of the latest release, so you'll have to install master - e.g. with

```
pip install --user git+https://github.com/blue-yonder/bonfire.git@master
```

Perhaps we should push a new release.... 

In the meantime, try the following curl with api in the url just to be sure that my hunch is right -

```
curl -Lv -H 'Accept: application/json' http://x1.devfactory.com:13912/api/users/<USERNAME>
```

let me know how it goes!
